src/core/dbus-mount.c

```diff
@@ -98,8 +100,30 @@ static int bus_mount_set_transient_property(
         if (streq(name, "Where"))
                 return bus_set_transient_path(u, name, &m->where, message, flags, error);
 
-        if (streq(name, "What"))
-                return bus_set_transient_string(u, name, &m->parameters_fragment.what, message, flags, error);
+        if (streq(name, "What")) {
+                _cleanup_free_ char *path = NULL;
+                const char *v;
+
+                r = sd_bus_message_read(message, "s", &v);
+                if (r < 0)
+                        return r;
+
+                if (!isempty(v)) {
+                        path = fstab_node_to_udev_node(v);
+                        if (!path)
+                                return -ENOMEM;
+
+                        if (strlen(path) >= PATH_MAX)
```

Comment:
path_is_valid()

Reply:
Hmm, I intentionally restricted the length only, because the source is not necessarily something we understand, and we pass it as-is to `mount`. Before the change we just used the plain `config_parse_string`.

OTOH I think if something longer than `PATH_MAX` is specified you're probably doing something wrong, hence this is added. I'm happy to drop this too if it doesn't make sense.

Reply:
it does make sense I guess, but please add a comment...